sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ManifestFileCommitProtocol.scala

```diff
@@ -40,14 +40,14 @@ class ManifestFileCommitProtocol(jobId: String, path: String)
   // Track the list of files added by a task, only used on the executors.
   @transient private var addedFiles: ArrayBuffer[String] = _
 
-  @transient private var fileLog: FileStreamSinkLog = _
+  @transient private var fileLog: MetadataLog[Array[SinkFileStatus]] = _
```

Comment:
This is necessary to not having branch on mode, but we can still apply `asInstanceOf` (with `require`) and keep this as it is. Please let me know what's our preference.